packages/components/src/text-input/src/TextInput.tsx

```diff
@@ -129,18 +156,21 @@ export function InnerTextInput(props: InnerTextInputProps) {
         placeholder,
         readOnly,
         required,
+        size: sizeValue,
         type,
         validationState,
         value: inputValue
     });
 
     const { hasFocus, inputProps: inputFocusProps } = useInputHasFocus();
 
-    const iconMarkup = useInputIcon(icon);
+    const iconMarkup = useInputIcon(icon, {
+        size: iconSize(sizeValue)
+    });
 
-    const buttonMarkup = useInputButton(button, !disabled && !readOnly);
+    const buttonMarkup = useInputButton(button, !disabled && !readOnly, { size: buttonSize(sizeValue) });
```

Comment:
i think this should be size: sizeValue. The useInputButton already do the conversion